addons/purchase/purchase.py

```diff
@@ -1078,7 +1087,8 @@ class MailComposeMessage(models.Model):
 
     @api.multi
     def send_mail(self, auto_commit=False):
-        if self._context.get('default_model') == 'purchase.order' and self._context.get('default_res_id'):
+        compose_internal = self.filtered('subtype_id.internal')
```

Comment:
we used  "message_post_with_view" method to render template and send mail which finally use send_mail method and this send_mail method is already overwriten in purchase,sale etc which update draft state. to send..

see send mail method in purchase.
https://github.com/odoo/odoo/blob/master/addons/purchase/purchase.py#L1064

problem is, when po is generated from procurement , this "make_po" method in purchase  will first find all draft state Po if no draft created then create new po. but in our case when send mail called from message_post_with_tempalte, this method update draft po to send . and
everytime make_po method call new po eventhough we have already related purchase order created.  so we avoid to update state when we have "internal subtype".

see make_po method.
https://github.com/odoo/odoo/pull/11612/files#diff-fe8587929089e5fdb6c75687011b1c7eR926